src/num.rs

```diff
@@ -461,14 +463,21 @@ macro_rules! float_impl {
                 x * x * (3 as Self - 2 as Self * x)
             }
 
+            fn smootherstep(e0: Self, e1: Self, t: Self) -> Self {
```

Comment:
This could be wrapped into a single statement and then you can do away with the `returns`:

```rust
if t < e0 {  
    Self::zero()
}
else if (t >= e1) { 
    Self::one() 
}
else {
    let x = (t - e0) / (e1 - e0);
    x * x * x * (3 as Self * x * (2 as Self * x - 5 as Self) + 10 as Self)
}
```